### PR TITLE
Fix retrocompatibilty issue with User - in Jira Software API there is…

### PR DESCRIPTION
**Pull request by oscarsj**
… Fix retrocompatibilty issue with User - in Jira Software API there is no accountId, we need to use "key" to identify an user. With this patch the code is compatible both with Jira Cloud (using "accountId") and Jira Software (using "key")

### CODE REVIEW

File: jira/client.py [hunk 1]
```diff
@@ -1567,6 +1567,17 @@ def _get_user_accountid(self, user):
...
         return accountId
 
+    def _get_user_key(self, user):
+        """Internal method for translating an user to a key.
+            Needed for backwards compatibilty with Jira Software API, not using
+            accountId
+        """
+        try:
+            key = self.search_users(user, maxResults=1)[0].key
+        except Exception as e:
```
Comment (studioj) [inline jira/client.py:1577]: would it be possible to specify the exception type you're catching here

**Comment by studioj** [comment]
could it be its related to https://github.com/pycontribs/jira/pull/940 ?

**Comment by adehad** [comment]
I think we may close this if this is handled better by #1053

**Comment by studioj** [comment]
asume fixed by #1053 => if not please reopen

**Pull request closed without merging**